lib/std/Build.zig

```diff
@@ -1532,24 +1532,25 @@ pub fn addUserInputFlag(b: *Build, name_raw: []const u8) !bool {
 }
 
 fn typeToEnum(comptime T: type) TypeId {
-    return switch (T) {
+    const simple_type: ?TypeId = switch (T) {
+        bool => .bool,
+        []const u8 => .string,
+        []const []const u8 => .list,
         std.zig.BuildId => .build_id,
-        else => return switch (@typeInfo(T)) {
-            .Int => .int,
-            .Float => .float,
-            .Bool => .bool,
-            .Enum => .@"enum",
-            .Pointer => |pointer| switch (pointer.child) {
-                u8 => .string,
-                []const u8 => .list,
-                else => switch (@typeInfo(pointer.child)) {
-                    .Enum => .enum_list,
-                    else => @compileError("Unsupported type: " ++ @typeName(T)),
-                },
-            },
-            else => @compileError("Unsupported type: " ++ @typeName(T)),
-        },
+        else => null,
     };
+    if (simple_type) |result| return result;
+
+    const complex_type: ?TypeId = switch (@typeInfo(T)) {
+        .Int => .int,
+        .Float => .float,
```

Comment:
Can you also add the `comptime` variants here? Tihs would allow using `.{ .foo = 10 }`

Reply:
This requires turning those values into strings, as they would be if they were passed through command line arguments; I made it into a separate PR since the code that needs to be changed is in another function: https://github.com/ziglang/zig/pull/20499